concordium-consensus/tests/scheduler/SchedulerTests/Payday.hs

```diff
@@ -74,6 +107,181 @@ testDoMintingP4 = do
+               in (uncurry KeyPairEd25519 k, addr) : ks g''
+
+numAccounts :: Int
+numAccounts = 10
+
+initialAccounts :: (IsProtocolVersion pv) => Accounts pv
+initialAccounts = foldr addAcc emptyAccounts (take numAccounts staticKeys)
+    where
+        addAcc (kp, addr) = putAccountWithRegIds (mkAccount (correspondingVerifyKey kp) addr initBal)
+        initBal = 10^(12::Int) :: Amount
+
+initialPureBlockState :: (IsProtocolVersion pv) => HashedBlockState pv
+initialPureBlockState = Concordium.GlobalState.Basic.BlockState.hashBlockState $ createBlockState initialAccounts
+
+initialPersistentBlockState :: (IsProtocolVersion pv, MonadBlobStore m) => m (HashedPersistentBlockState pv)
+initialPersistentBlockState = makePersistent . _unhashedBlockState $ initialPureBlockState
+
+genesis :: (IsProtocolVersion pv) => Word -> (GenesisData pv, [(BakerIdentity, FullBakerInfo)], Amount)
+genesis nBakers =
+    makeGenesisData
+    0
+    nBakers
+    1000
+    defaultFinalizationParameters
+    dummyCryptographicParameters
+    emptyIdentityProviders
+    dummyArs
+    []
+    1234
+    dummyKeyCollection
+    dummyChainParameters
+
+-- this is, perhaps, the smallest effect type that makes these tests run
+
+type MyPureBlockState pv = HashedBlockState pv
+type MyPureTreeState pv = SkovData pv (MyPureBlockState pv)
+type MyPureMonad pv = PureTreeStateMonad (MyPureBlockState pv) (PureBlockStateMonad pv (StateT (MyPureTreeState pv) IO))
+
+runMyPureMonad :: (IsProtocolVersion pv) => MyPureTreeState pv -> MyPureMonad pv a -> IO (a, MyPureTreeState pv)
+runMyPureMonad is = (`runStateT` is) . runPureBlockStateMonad . runPureTreeStateMonad
+
+runMyPureMonad' :: (IsProtocolVersion pv) => GenesisData pv -> MyPureMonad pv a -> IO (a, MyPureTreeState pv)
+runMyPureMonad' gd = runPureBlockStateMonad (initialSkovDataDefault gd initialPureBlockState) >>= runMyPureMonad
+
+type MyPersistentBlockState pv = HashedPersistentBlockState pv
+type MyPersistentTreeState pv = SkovPersistentData pv () (MyPersistentBlockState pv)
+type MyPersistentMonad pv = PersistentTreeStateMonad () (MyPersistentBlockState pv) (PersistentBlockStateMonad pv BlobStore (ReaderT BlobStore (StateT (MyPersistentTreeState pv) LogIO)))
+
+fakeLogMethod :: LogMethod IO
+fakeLogMethod _ _ _ = return ()
+
+runMyPersistentMonad :: (IsProtocolVersion pv) => BlobStore -> MyPersistentTreeState pv -> MyPersistentMonad pv a -> IO (a, MyPersistentTreeState pv)
+runMyPersistentMonad bSt is = (`runLoggerT` fakeLogMethod) . (`runStateT` is) . (`runReaderT` bSt) . runPersistentBlockStateMonad . runPersistentTreeStateMonad
+
+runMyPersistentMonad' :: (IsProtocolVersion pv) => GenesisData pv -> MyPersistentMonad pv a -> IO (a, MyPersistentTreeState pv)
+runMyPersistentMonad' gd mpm = do
+  nonce <- randomIO :: IO Int
+  let dbpath = "./testdb_" ++ show nonce ++ "/"
+  let treestatepath = dbpath ++ "treestate/"
+  let blockstatepath = dbpath ++ "blockstate.dat"
+  myBlobStore <- createBlobStore blockstatepath
```

Comment:
Won't this leave random files in the directory from which the tests are run. That is not ideal.